tests/test_resolver.py

```diff
@@ -349,3 +352,32 @@ def test_resolvable_set_built():
   updated_rs.merge(rq, [binary_pkg])
   assert updated_rs.get('foo') == set([binary_pkg])
   assert updated_rs.packages() == [(rq, set([binary_pkg]), None, False)]
+
+def test_resolver_blacklist():
+  project = make_sdist(name='project', version='1.0.0')
```

Comment:
this probably needs a `setup.py`-sourced transitive dep to blacklist as a complete test case?